dashboard/app/models/user_level.rb

```diff
@@ -138,15 +130,31 @@ def before_unsubmit
     end
   end
 
-  def has_autolocked?(stage)
-    return false unless stage.lockable?
-    unlocked_at && unlocked_at < AUTOLOCK_PERIOD.ago
+  # `locked` is a virtual attribute because it relies on `unlocked_at` to
```

Comment:
I think it's possible to make unlocked_at private if you'd like to enforce that external callers not use it.

It might also be good to include some of the explanation of why we needed to use unlocked_at, either here or where you make unlocked_at private.

Reply:
I'm making this private now, though it might take a day to rework the tests to get this operational. Thanks for the suggestion!